src/systems/systems/src/pneumatic/mod.rs

```diff
@@ -75,12 +143,38 @@ impl PneumaticContainer for PneumaticPipe {
         self.temperature
     }
 
-    // Adds or removes a certain amount of air
-    fn change_fluid_amount(&mut self, volume_change: Volume) {
-        let pressure_change = self.calculate_pressure_change_for_volume_change(volume_change);
+    fn mass(&self) -> Mass {
+        /*
+        We calculate the mass instead of storing it because floating points are
+        not exact and after some time the true mass drifts from the calculated
+        mass which then results in some negative pressures and temperatures
+        when doing heat conduction. This later leads to NaNs.
+        Another possibility would be to calculate the volume instead.
+        */
```

Comment:
Interesting observation 👍 What would you think about eliminating `pressure` instead and computing it from mass, volume and temperature? In my mind, using `mass` as a form of "source of truth", rather than pressure makes more sense for some reason.

Reply:
To be honest I looked at the performance perspective to decide which component to calculate. I simply checked how often the mass is queried. Both pressure and temperature are queried relatively often (also for all the simvars). But I agree with you to better calculate something else. Maybe we can remove the volume? It's not used anymore anyway.

Reply:
As a side note I'd like to mention that comments in this PR aren't following the Rust convention. If you'd like to know more, please read [this chapter](https://doc.rust-lang.org/book/ch03-04-comments.html) of the book and [this section](https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#making-useful-documentation-comments) as well.

Reply:
> As a side note I'd like to mention that comments in this PR aren't following the Rust convention. If you'd like to know more, please read [this chapter](https://doc.rust-lang.org/book/ch03-04-comments.html) of the book and [this section](https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#making-useful-documentation-comments) as well.

Thank you for your comment. I didn't know there is a convention for comments in Rust. I will adopt them. But what I don't understand is why only line comments are convention but that's another topic.